src/wp-includes/template.php

```diff
@@ -711,7 +711,25 @@ function locate_template( $template_names, $load = false, $require_once = true,
 	}
 
 	if ( $load && '' !== $located ) {
+		/**
+		 * Fires before the located template is loaded
+		 *
+		 * @since 6.0.0
+		 *
+		 * @param string The template filename.
+		 */
+		do_action( 'before_load_template', $located );
```

Comment:
```suggestion
		do_action( 'wp_before_load_template', $located );
```

The prefix will help avoid any naming collisions with plugins and themes.

🔢 Applies to the after hook's name too.